website/api-gateway/src/kafka/kafka-events.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  AddMetaParams,
+  FindMessageParams,
+  FindProgramParams,
+  GetAllProgramsParams,
+  GetIncomingMessagesParams,
+  GetMessagesParams,
+  GetMetaParams,
+  GetOutgoingMessagesParams,
+  GetTestBalanceParams,
+  GetAllUserProgramsParams,
+} from '@gear-js/interfaces';
+import { nanoid } from 'nanoid';
+
+import { kafkaProducer, kafkaSendByTopic } from './kafka';
+import { KAFKA_TOPICS, kafkaProducerTopics } from '../common/kafka-producer-topics';
+import { deleteKafkaEvent, kafkaEventMap } from './kafka-event-map';
+
+async function programData(params: FindProgramParams) {
+  const correlationId: string = nanoid(6);
+  await kafkaSendByTopic(KAFKA_TOPICS.PROGRAM_DATA, correlationId, params);
+
+  let topicEvent;
+  const res = new Promise((resolve) => (topicEvent = resolve));
+  kafkaEventMap.set(correlationId, topicEvent);
+  deleteKafkaEvent(correlationId);
```

Comment:
What happens if response will be received later than 3 seconds?

Reply:
My mistake, fixed this part. Deleted from all `Kafka` event features `deleteKafkaEvent`.

Moved `deleteKafkaEvent` to `kafkaConsumer.run`

Now it looks like below. 

```javascript
await kafkaConsumer.run({
      eachMessage: async ({ message }) => {
        const correlationId = message?.headers?.kafka_correlationId.toString();
        const resultFromService = kafkaEventMap.get(correlationId);
        if (resultFromService) await resultFromService(JSON.parse(message.value.toString()));
        deleteKafkaEvent(correlationId);
      },
    });
```